lib/config/fileconf.go

```diff
@@ -512,8 +512,10 @@ type ConnectionRate struct {
 // ConnectionLimits sets up connection limiter
 type ConnectionLimits struct {
 	MaxConnections int64            `yaml:"max_connections"`
-	MaxUsers       int              `yaml:"max_users"`
 	Rates          []ConnectionRate `yaml:"rates,omitempty"`
+
+	// MaxUsers is deprecated and has no effect.
+	MaxUsers int `yaml:"max_users"`
```

Comment:
For IDE/gopls deprecation indicators:

```suggestion
	// Deprecated: MaxUsers has no effect.
	MaxUsers int `yaml:"max_users"`
```

or maybe we could even declare it as a blank identifier instead?

```suggestion
	// The max_users field is deprecated and has no effect.
	_ int `yaml:"max_users"`
```